Model/CustomItemExportSchedulerModel.php

```diff
@@ -0,0 +1,294 @@
+
+                    if (0 == count($results) && $customItemAdded) {
+                        break;
+                    }
+
+                    if (count($results) < self::CONTACT_LIMIT) {
+                        $fetchResult = false;
+                    } else {
+                        $contactOffset += self::CONTACT_LIMIT;
+                    }
+
+                    $rowData   = $savedRow;
+                    $rowData[] = implode(',', $results);
+
+                    if ($this->eventDispatcher->hasListeners(CustomItemEvents::ON_PROCESSING_FILE)) {
+                        $this->eventDispatcher->dispatch(CustomItemEvents::ON_PROCESSING_FILE);
+                    }
+
+                    fputcsv($handler, $rowData);
+                    $customItemAdded = true;
+                }
+            }
+        }
+
+        fclose($handler);
+    }
+
+    /**
+     * @param array<CustomField> $customFields
+     */
+    private function addExportFileHeaderToCsvFile(array $customFields): void
+    {
+        $header  = $this->getCSVHeader($customFields);
+        $handler = @fopen($this->filePath, 'ab+');
+        fputcsv($handler, $header);
+        fclose($handler);
+    }
+
+    /**
+     * @return array<int>
+     */
+    private function getContactIds(CustomItem $customItem, int $limit = 200, int $offset = 0): array
+    {
+        $contactIds = $this->customItemXrefContactRepository
+            ->getContactIdsLinkedToCustomItem($customItem->getId(), $limit, $offset);
+
+        return array_column($contactIds, 'contact_id');
+    }
+
+    public function sendEmail(CustomItemExportScheduler $customItemExportScheduler, string $filePath): void
+    {
+        $user = $customItemExportScheduler->getUser();
+
+        $message = $this->getEmailMessageWithLink($filePath);
+        $this->mailHelper->setTo([$user->getEmail() => $user->getName()]);
+        $this->mailHelper->setSubject(
+            $this->translator->trans('custom.item.export.email_subject', ['%file_name%' => basename($filePath)])
+        );
+        $this->mailHelper->setBody($message);
+        $this->mailHelper->parsePlainText($message);
+        $this->mailHelper->send(true);
+    }
+
+    public function getEmailMessageWithLink(string $filePath): string
+    {
+        $link = $this->customItemRouteProvider->buildExportDownloadRoute(basename($filePath));
+
+        return $this->translator->trans(
+            'custom.item.export.email',
+            ['%link%' => $link, '%label%' => basename($filePath)]
+        );
+    }
+
+    /**
+     * @throws \Doctrine\ORM\ORMException
+     * @throws \Doctrine\ORM\OptimisticLockException
+     */
+    public function deleteEntity(CustomItemExportScheduler $customItemExportScheduler): void
+    {
+        $this->em->remove($customItemExportScheduler);
+        $this->em->flush();
+    }
+
+    public function getExportFileToDownload(string $fileName): BinaryFileResponse
```

Comment:
Responses are fairly high in the layer stack. A model is not the place for them. I would expect to see them almost exclusively in controllers.